ros_buildfarm/colcon/verb/scope.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from colcon_core.package_identification.ignore import IGNORE_MARKER
+from colcon_core.package_selection import add_arguments \
+    as add_packages_arguments
+from colcon_core.package_selection import get_packages
+from colcon_core.plugin_system import satisfies_version
+from colcon_core.verb import VerbExtensionPoint
+
+
+class ScopeVerb(VerbExtensionPoint):
+    """Scope workspace to a pool of packages by ignoring unrelated ones."""
+
+    def __init__(self):
+        super().__init__()
+        satisfies_version(VerbExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
+
+    def add_arguments(self, *, parser):
+        parser.add_argument(
+            'depth_before', type=int, metavar='NUM_BEFORE',
+            help='Number of forward-dependent packages upon which ' +
+                 'the targeted package(s) depends.')
```

Comment:
I'd guess I don't really have a specific use case. Maybe I just misinterpreted something from our discussion on the desired package selection mechanisms for this feature.

If you see it as cruft, I can remove it, but I'll have to plumb a new mechanism to use use `--packages-up-to` instead, since some use cases involve "islands" of packages built on top of an underlay, and others involve building up from scratch.

Reply:
While I dislike the problems of `depth_after` I see when that is useful (build downstream packages to check if they still pass their tests after changes in a set of packages).

For `depth_before` I just don't see any use case. I don't see when "islands" of packages would be described by the depth before. If we don't have a case we want to use it for I would rather remove it.